### PR TITLE
feat: Remove app-pods-secure scrape job

### PR DESCRIPTION
**Pull request by shorim**
## Description

Changes proposed in this pull request (what was done and why):

- Remove scape job `app-pods-secure` which was responsible for scraping annotated Pods that either have an Istio sidecar or have the "prometheus.io/scheme" annotation set to "https"
- Refactor the functions responsible for generating configs for prometheus receiver
- Adjust documentation
- Adjust unit tests
- Add an e2e test

Changes refer to particular issues, PRs or documents:

- #1468

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: docs/user/04-metrics.md [hunk 1]
```diff
@@ -313,14 +313,16 @@ spec:
...
-| `prometheus.io/path`               | `/metrics`, `/custom_metrics` | `/metrics` | Defines the HTTP path where Prometheus can find metrics data.                                                                                                                                                                                                                                                                               |
-| `prometheus.io/scheme`             | `http`, `https` | If Istio is active, `https` is supported; otherwise, only `http` is available. The default scheme is `http` unless an Istio sidecar is present, denoted by the label `security.istio.io/tlsMode=istio`, in which case `https` becomes the default. | Determines the protocol used for scraping metrics — either HTTPS with mTLS or plain HTTP. |
+For metrics ingestion to start automatically:
```
Comment (NHingerl) [inline docs/user/04-metrics.md:316]:
```suggestion
For metrics ingestion to start automatically, use the annotations of the following table. If Istio is active (recommended), apply them to a Service that resolves your metrics port. Only if Istio is inactive, you can alternatively apply the annotations directly to the Pod.
```
I assume that for most users, Istio is going to be active?
Or: What do you mean with "Istio is active" - the Istio module is added to the cluster (by default, yes)? Or Istio sidecar injection is enabled (by default, nope)?

File: docs/user/04-metrics.md [hunk 2]
```diff
@@ -313,14 +313,16 @@ spec:
...
-| `prometheus.io/path`               | `/metrics`, `/custom_metrics` | `/metrics` | Defines the HTTP path where Prometheus can find metrics data.                                                                                                                                                                                                                                                                               |
-| `prometheus.io/scheme`             | `http`, `https` | If Istio is active, `https` is supported; otherwise, only `http` is available. The default scheme is `http` unless an Istio sidecar is present, denoted by the label `security.istio.io/tlsMode=istio`, in which case `https` becomes the default. | Determines the protocol used for scraping metrics — either HTTPS with mTLS or plain HTTP. |
+For metrics ingestion to start automatically: 
+* If istio is not active, apply the annotations in the table below either to a Service that resolves your metrics port, or directly to the Pod.
```
Comment (NHingerl) [inline docs/user/04-metrics.md:317]:
```suggestion
```

File: docs/user/04-metrics.md [hunk 3]
```diff
@@ -313,14 +313,16 @@ spec:
...
+* If istio is active, apply the annotations in the table below to a Service that resolves your metrics port. Applying the annotations directly to the Pod will not trigger metrics scraping if istio is active.
```
Comment (NHingerl) [inline docs/user/04-metrics.md:318]: I believe this is the default case, right? So I'd put it first.

```suggestion
```

File: docs/user/04-metrics.md (hunk 1 above)
Comment (shorim) [reply docs/user/04-metrics.md:316]: with "istio is active", I meant the Istio sidecar is present. I will change the wording there to make it more clear

**Review by NHingerl — DISMISSED** [review]

**Review by skhalash — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**